base/linalg.jl

```diff
@@ -176,10 +176,11 @@ else
 end
 
 # Check that stride of matrix/vector is 1
-function chkstride1(A...)
-    for a in A
-        stride(a,1)== 1 || error("matrix does not have contiguous columns")
-    end
+@inline _chkstride1(a) = stride(a,1) == 1 || error("matrix does not have contiguous columns")
+
+macro chkstride1(A...)
+    @assert isa(A, Tuple)
+    return Expr(:block, [:(_chkstride1($(esc(A[i])))) for i in 1:length(A)]...)
```

Comment:
I guess one advantage is that you can combine the check and have only one `error()`